### PR TITLE
Add some methods to remove and add roles

### PR DESCRIPTION
**Pull request by mcnijman**
Hi, I would like to be able to remove roles and remove a user from the database which isn't possible without removing the role relations.

These methods are from the doctrine ArrayCollection: 
http://www.doctrine-project.org/api/common/2.1/class-Doctrine.Common.Collections.ArrayCollection.html

### CODE REVIEW

File: src/SamUser/Entity/User.php [hunk 1]
```diff
@@ -230,6 +231,44 @@ public function getRoles()
...
+     * Add roles to the user.
+     *
+     * @param Collection $roles
+     *
+     * @return void
+     */
+    public function addRoles(Collection $roles)
+    {
+        foreach ($roles as $role)
+            $this->roles->add($role);
```
Comment (manuakasam) [inline src/SamUser/Entity/User.php]: Even though it's valid PHP wise, this doesn't go hand in hand with PSR-2. Please use braces even if the body is just one line ;) https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#55-foreach

File: src/SamUser/Entity/User.php [hunk 2]
```diff
@@ -230,6 +231,44 @@ public function getRoles()
...
+     * Remove roles from the user.
+     *
+     * @param Collection $roles
+     *
+     * @return void
+     */
+    public function removeRoles(Collection $roles)
+    {
+        foreach ($roles as $role)
+            $this->roles->removeElement($role);
```
Comment (manuakasam) [inline src/SamUser/Entity/User.php]: same as above

**Comment by manuakasam** [comment]
This can easily be merged once you've changed the CS for the foreach blocks.

File: src/SamUser/Entity/User.php [hunk 3]
```diff
@@ -230,6 +231,44 @@ public function getRoles()
...
+        $this->roles->add($role);
+    }
+    
+    /**
+     * Remove a role from the user.
+     *
+     * @param Role $role
+     *
+     * @return void
+     */
```
Comment (manuakasam) [inline src/SamUser/Entity/User.php]: I did this as well, but is there any chance you could remove all "return void" docblocks? "return void" is bad :) Was wrong of me in the first place tho ;) not your bad ^^

Comment (mcnijman) [inline src/SamUser/Entity/User.php]: Sure, i'll change the foreach block. I usually return the entity (return $this;) on set / add & remove. Shall I put that in or just remove the "return void"?

**Comment by manuakasam** [comment]
Thank you for your contribution @mcnijman

**Comment by mcnijman** [comment]
Thanks for the quick replies & merge!

**Pull request merged**